src/main/java/io/crate/testing/CrateTestCluster.java

```diff
@@ -134,7 +139,12 @@ public Builder keepWorkingDir(boolean keepWorkingDir) {
         }
 
         public CrateTestCluster build() {
-            Preconditions.checkArgument(clusterName != null, "no cluster name given");
+            if (clusterName == null) {
+                throw new IllegalArgumentException("no cluster name is given");
+            }
+            if (numberOfNodes <= 0) {
```

Comment:
can you check such stuff in the setter, this is more convinient for the user to find the cause